src/lakefs_spec/spec.py

```diff
@@ -427,7 +427,7 @@ def _ls_from_cache(self, path: str, recursive: bool = False) -> list[dict[str, A
                 result = []
             result.extend(files)
         if not result:
-            return result
+            return None
```

Comment:
This change incurs an additional API call when the dircache returns an empty directory (`result == []`).

I would revert back to the old version, or make the intent more explicit by changing the condition to `if result is None`.

Reply:
Yes, I forgot. It's the first attempt to fix the cache problem.